packages/react-native/scripts/cocoapods/codegen.rb

```diff
@@ -11,8 +11,9 @@
 # - dir_manager: a class that implements the `Dir` interface. Defaults to `Dir`, the Dependency can be injected for testing purposes.
 # @throws an error if it could not find the codegen folder.
 def build_codegen!(react_native_path, relative_installation_root, dir_manager: Dir)
-    codegen_repo_path = "#{relative_installation_root}/#{react_native_path}/../react-native-codegen";
-    codegen_npm_path = "#{relative_installation_root}/#{react_native_path}/../@react-native/codegen";
+    is_react_native_path_absolute = react_native_path.start_with?("/")
+    codegen_repo_path = "#{is_react_native_path_absolute ? '' : relative_installation_root.to_s + '/'}#{react_native_path}/../react-native-codegen";
+    codegen_npm_path = "#{is_react_native_path_absolute ? '' : relative_installation_root.to_s + '/'}#{react_native_path}/../@react-native/codegen";
```

Comment:
can we change the implementation a little?
I would do something like this:

```suggestion
def basePath(react_native_path, relative_installation_root)
  if react_native_path.start_with?("/")
    react_native_path
  else 
    "#{relative_installation_root.to_s}/#{react_native_path}"
end

def build_codegen!(react_native_path, relative_installation_root, dir_manager: Dir)
  codegen_repo_path = "#{basePath(react_native_path, relative_installation_root)}/../react-native-codegen"
  codegen_npm_path = "#{basePath(react_native_path, relative_installation_root)}/../@react-native/codegen"
```

This allow us to reuse this function down the line like this:
```
'node',
        [
            "#{basePath(react_native_path, relative_installation_root)}/scripts/generate-provider-cli.js",
```
Reducing code duplication and logic duplication.

Reply:
updated